dpctl/memory/_memory.pxd

```diff
@@ -28,7 +28,7 @@ from .._sycl_device cimport SyclDevice
 from .._sycl_queue cimport SyclQueue
 
 
-cdef class _Memory:
+cdef public class _Memory [object Py_MemoryObject, type Py_MemoryType]:
```

Comment:
Do we need to expose the `_Memory` class?

Reply:
I thought yes. Mostly for type detection. Testing that an object `obj` is a USM memory buffer can be as simpler as using `PyObject_TypeCheck` with `Py_MemoryType`.